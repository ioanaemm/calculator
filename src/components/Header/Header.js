import React from "react";
import PropTypes from "prop-types";

export default function Header(props) {
  return <p className="result">{props.result}</p>;
}

Header.propTypes = {
  // value to be displayed
  result: PropTypes.string
};
