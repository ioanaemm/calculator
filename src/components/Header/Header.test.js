import React from "react";
import TestRenderer from "react-test-renderer";
import Header from "./Header";

describe("Header component", () => {
  it("exists", () => {
    const component = TestRenderer.create(<Header />);
    expect(component).toBeDefined();
  });

  it("works if it has no props", () => {
    const component = TestRenderer.create(<Header />);
    expect(component).toBeDefined();
  });

  it("correctly renders the result", () => {
    let result = TestRenderer.create(<Header result="12" />);
    expect(result).toMatchSnapshot();
  });
});
