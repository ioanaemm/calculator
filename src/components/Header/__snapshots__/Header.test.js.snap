// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component correctly renders the result 1`] = `
<p
  className="result"
>
  12
</p>
`;
